pkg/ingester/ingester.go

```diff
@@ -205,14 +209,22 @@ func New(cfg Config, clientConfig client.Config, limits *validation.Overrides, c
 	}
 
 	var err error
-	i.lifecycler, err = ring.NewLifecycler(cfg.LifecyclerConfig, i, "ingester")
+	i.limiter = NewSeriesLimiter(limits, i.lifecycler, cfg.LifecyclerConfig.RingConfig.ReplicationFactor, cfg.ShardByAllLabels)
+
+	i.wal, err = newWAL(cfg.WALConfig, i)
+	if err != nil {
+		return nil, err
+	}
+
+	i.lifecycler, err = ring.NewLifecycler(cfg.LifecyclerConfig, i, "ingester", !cfg.WALConfig.walEnabled)
 	if err != nil {
 		return nil, err
 	}
 
 	// Init the limter and instantiate the user states which depend on it
```

Comment:
We're not initializing the limiter here anymore (you moved it), so you should adapt comments, while keeping the same level of information.

Reply:
Oops, I missed the comments. The user states are actually also created when recovering from WAL. I will modify it to make it more readable.